sdk/cdn/Azure.ResourceManager.Cdn/src/Generated/Models/DeliveryRuleRequestHeaderAction.cs

```diff
@@ -10,7 +10,7 @@
 namespace Azure.ResourceManager.Cdn.Models
 {
     /// <summary> Defines the request header action for the delivery rule. </summary>
-    public partial class DeliveryRuleRequestHeaderAction : DeliveryRuleActionAutoGenerated
+    internal partial class DeliveryRuleRequestHeaderAction : DeliveryRuleActionAutoGenerated
```

Comment:
I am curious why these are hidden.  It seems that mgmt plane never uses ResponseWithHeaders as a return type and I am not sure if that is by design.  We need to investigate why this is before we hide these .

Reply:
Good catch - but I am not sure if this is related with the `ResponseWithHeaders`.
This class should be public because this is one of the derived class with discriminator. I will update the implementation to make sure we could handle this case. Here is the corresponding swagger of this: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json#L4533